pandas/core/series.py

```diff
@@ -1117,27 +1117,24 @@ def count(self, level=None):
         -------
         nobs : int or Series (if level specified)
         """
-        if level is not None:
-            mask = notnull(self.values)
+        from pandas.core.index import _get_na_value
 
-            if isinstance(level, compat.string_types):
-                level = self.index._get_level_number(level)
+        if level is None:
+            return notnull(_values_from_object(self)).sum()
 
-            level_index = self.index.levels[level]
+        if isinstance(level, compat.string_types):
+            level = self.index._get_level_number(level)
 
-            if len(self) == 0:
-                return self._constructor(0, index=level_index)\
-                           .__finalize__(self)
+        lev = self.index.levels[level]
+        lab = np.array(self.index.labels[level], subok=False, copy=True)
 
-            # call cython function
-            max_bin = len(level_index)
-            labels = com._ensure_int64(self.index.labels[level])
-            counts = lib.count_level_1d(mask.view(np.uint8),
-                                        labels, max_bin)
-            return self._constructor(counts,
-                                     index=level_index).__finalize__(self)
+        mask = lab == -1
+        if mask.any():
+            lab[mask] = cnt = len(lev)
```

Comment:
my point is why we having a special case implementation, when simply using ``s.groupby(level=level).count()`` is acceptable? 

and this is what all of the other ``make_stat_*`` functions do.

Since this was a bug nothing is even being eliminated.